### PR TITLE
Support partitioning hints for athena iceberg

### PR DESCRIPTION
**Pull request by steinitzu**
### Description
Adds `athena_adapter` to add partitioning hints in athena. Generates `PARTITIONED BY` clause on iceberg tables.

The adapter generates a table hint: `x-athena-partition`.
This is a dict in form of `{<column_name>: <template_string>}`  
E.g. `{"department": "{column_name}", "date_hired": "year({column_name})"}`  

Using format strings so we can generate transform (`bucket(...)`, `year(...)`, etc) strings and inject the escaped column name later in the loader.

### Related Issues

Implements partially #555 -> Iceberg support

### CODE REVIEW

File: tests/load/athena_iceberg/test_athena_adapter.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import pytest
+
+import dlt
+from dlt.destinations import filesystem
+from dlt.destinations.impl.athena.athena_adapter import athena_adapter, athena_partition
+from tests.load.utils import destinations_configs, DestinationTestConfiguration
+
+
+def test_iceberg_partition_hints():
```
Comment (rudolfix) [inline tests/load/athena_iceberg/test_athena_adapter.py:9]: should we use `destination_config` to filter for `athena` so we run it only when athena tests run on CI?

File: tests/load/pipeline/test_athena.py [hunk 2]
```diff
@@ -231,3 +233,68 @@ def test_athena_file_layouts(destination_config: DestinationTestConfiguration, l
...
     )
     assert table_counts == {"items1": 3, "items2": 7}
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(default_sql_configs=True, subset=["athena"], force_iceberg=True),
+    ids=lambda x: x.name,
+)
+def test_athena_partitioned_iceberg_table(destination_config: DestinationTestConfiguration):
```
Comment (rudolfix) [inline tests/load/pipeline/test_athena.py:243]: please mark this test with `essentials` so we run it on a each CI run

**Review by rudolfix — CHANGES_REQUESTED** [review]
code and docs: LGTM!
I think we can reconfigure tests, see my review

File: tests/load/athena_iceberg/test_athena_adapter.py (hunk 1 above)
Comment (steinitzu) [reply tests/load/athena_iceberg/test_athena_adapter.py:9]: The whole package is skipped by: https://github.com/dlt-hub/dlt/blob/athena-partitions/tests/load/athena_iceberg/__init__.py#L1-L4

File: tests/load/pipeline/test_athena.py (hunk 2 above)
Comment (steinitzu) [reply tests/load/pipeline/test_athena.py:243]: The module is marked https://github.com/dlt-hub/dlt/blob/athena-partitions/tests/load/pipeline/test_athena.py#L23-L24

Comment (steinitzu) [reply tests/load/pipeline/test_athena.py:243]: Or was this meant for the `athena_adapter` test?

Comment (rudolfix) [reply tests/load/pipeline/test_athena.py:243]: heh I meant here... so it seems that all is good

**Pull request merged**